src/GameSelector/index.tsx

```diff
@@ -1,70 +1,146 @@
 import { NativeStackScreenProps } from '@react-navigation/native-stack';
-import { Button, Center, FormControl, Input, VStack } from 'native-base';
-import React from 'react';
-import { RootStackParamList } from '../../App';
+import {
+  Box,
+  Button,
+  Center,
+  FormControl,
+  HStack,
+  Input,
+  View,
+  VStack,
+  Text,
+  CheckIcon,
+  Flex,
+  SmallCloseIcon,
+  Stack,
+  ScrollView,
+  Heading,
+} from 'native-base';
+import React, { useEffect, useState } from 'react';
+import { API_URL } from '../constant';
+import { Board } from '../GameState/Board';
+import { Game } from '../interface';
 
 export interface GameSelectorFormData {
   gameId: number;
 }
 
-type Props = NativeStackScreenProps<RootStackParamList, 'GameSelector'>;
+export default function ({ navigation, player }: any) {
+  const [userGames, setUserGames] = useState<Game[]>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errorUserGames, setErrorUserGames] = useState<any>();
 
-export default function ({ navigation }: Props) {
-  const [formData, setData] = React.useState<GameSelectorFormData>({
-    gameId: -1,
-  });
-  const [errors, setErrors] = React.useState({});
-
-  const validate = () => {
-    if (!formData.gameId) {
-      setErrors({
-        ...errors,
-        gameId: 'The game id cannot be empty',
-      });
-      return false;
+  const getUserGames = async () => {
+    try {
+      const response = await fetch(
+        `${API_URL}/games?filter=${encodeURI(
+          JSON.stringify({ playerNumber: [player.id] }),
+        )}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: 'Bearer ' + player.token,
+          },
+        },
+      );
+      const json = (await response.json()).data as Game[];
+      setUserGames(json);
+    } catch (error) {
+      setErrorUserGames(error);
+    } finally {
+      setIsLoading(false);
     }
+  };
 
-    setErrors({});
+  const navigateToGameState = (id: number) => {
     navigation.navigate('GameState', {
-      gameId: formData.gameId,
+      gameId: id,
     });
-    return true;
   };
 
-  const handleChangeText = (value: string) => {
-    setData({ ...formData, gameId: +value });
+  useEffect(() => {
+    getUserGames();
+  }, []);
+
+  const UserGamesList = () => {
+    return (
```

Comment:
What do you display in case there is an `errorUserGames` ?

Reply:
I just added a toast to warn if the fetch was not successful. 
And also a message if the player has no game in progress

Reply:
ok